src/playground-code-editor.ts

```diff
@@ -384,8 +407,29 @@ export class PlaygroundCodeEditor extends LitElement {
         this._applyHideAndFoldRegions();
         this._showDiagnostics();
       } else {
-        // Change event is only for user input.
-        this.dispatchEvent(new Event('change'));
+        const previousToken = _editorInstance.getTokenAt(changeObject.from);
+        const tokenUnderCursor = this.tokenUnderCursor.string.trim();
+        // To help reduce round trips to a language service or a completion provider, we
+        // are providing a flag if the completion is building on top of the earlier recommendations.
+        // If the flag is true, the completion system can just filter the already stored
+        // collection of completions again with the more precise input.
+        // On deletion events, we want to query the LS again, since we might be in a new context after
+        // removing characters from our code.
+        const isInputEvent = changeObject.origin === EditorChangeOrigin.INPUT;
+        const isCompletingCompletions =
+          (tokenUnderCursor.length > 1 || previousToken.string === '.') &&
```

Comment:
Could we lets the user provide a "completionTrigger" character or characters? So it would trigger on those, or when 2+ chars are input? And we could default the default trigger to . ?

Reply:
I've added a function that currently just checks if the type of the file is "ts" and doesn't ask for completions if it isn't. We can refine it down the line